machetli/pddl/generators.py

```diff
@@ -83,3 +83,40 @@ def get_successors(self, state):
                 visitors.TaskElementEraseObjectVisitor(name))
             yield Successor(child_state,
                             f"Removed object '{name}'. Remaining objects: {len(task.objects) - 1}")
+
+class RemoveInit(SuccessorGenerator):
```

Comment:
Removing a fact from init makes the task simpler in the sense that the PDDL is smaller but it might become harder to solve. Should we also add a generator that adds facts to the initial state? You couldn't use them both without running into an infinite loop but either one could be useful.